kong/plugins/prometheus/exporter.lua

```diff
@@ -168,6 +181,50 @@ local function collect()
                  "/metrics endpoint: ", err)
   end
 
+  -- erase all target/upstream metrics, prevent exposing old metrics
+  metrics.upstream_target_health:reset()
+
+  -- upstream targets accessible?
+  local upstreams_dict = balancer.get_all_upstreams()
```

Comment:
@gs and @locao, is this API changing with 2.1 refactors around balancers? Is it safe to use this in a plugin?

Reply:
@hbagdi yes, the keys of the dict have changed, they are now "ws_id:upstream_id"

Reply:
These balancer APIs were not intended to be public stable APIs, but since this is a bundled plugin we have control over, if you folks want to do this for now and ensure it's aligned with 2.1 (by fixing the extraction of the upstream_name as described above), it's fine by me, but for the future I would recommend replacing this with a new proper PDK-exposed stable function.

Reply:
@hishamhm is the alterations at branch "next" from Kong? 
I'm look the `load_upstreams_dict_into_memory` and it is setting the dict as `upstreams_dict[up.name] = up.id`. The function is used to populate the cache.

Reply:
@hishamhm is it already at "next" branch? I'm trying to

Reply:
oh! "up.name" has the pattern "ws_id:upstream_name" now ?
Do you think using: `local _, upstream_name = key:match("^([^:]+):(.-)$")`  would be good?

Reply:
or:
```
local _, upstream_name = key:match("^([^:]*):(.-)$")
if not upstream_name then upstream_name = key end
```
for compabilities

Reply:
I've submitted this code ^

Reply:
@carnei-ro Yes, it was a very recent change in `next` — your change LGTM! Thank you for the super quick change!